lib/matplotlib/axis.py

```diff
@@ -1965,7 +1966,11 @@ def set_ticklabels(self, labels, *, minor=False, fontdict=None, **kwargs):
             raise TypeError(f"{labels:=} must be a sequence") from None
         locator = (self.get_minor_locator() if minor
                    else self.get_major_locator())
-        if isinstance(locator, mticker.FixedLocator):
+        if not labels:
+            # eg labels=[]:
+            formatter = mticker.NullFormatter()
```

Comment:
This does not have an effect if you return in the next line. `formatter` is only used in lines 2005ff.

Reply:
Obviously, this is untested. You can add a test that there is no warning using

```
with warnings.catch_warnings():
    warnings.simplefilter("error")
```

https://docs.pytest.org/en/7.0.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests

Reply:
Oops thanks I knew there was a reason to not do that.  

It is tested, but the old code ends up being used

I'll move to draft as I'm away for a few days

Reply:
Is the warning filter necessary, since we run with warnings as errors in general?  C.f. the test added at #26300.

Reply:
It's not necessary but prevents the same warning appearing with two different syntaxes.